driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1473,6 +1476,36 @@ public Builder withCloudSecureConnectBundle(InputStream cloudConfigInputStream)
       return addCloudConfigToBuilder(cloudConfig);
     }
 
+    /**
+     * Disables advanced shard awareness. By default, this driver chooses local port while making a
+     * connection to node, to signal which shard it wants to connect to. This allows driver to
+     * estabilish connection pool faster, especially when there are multiple clients connecting
+     * concurrently. If this causes any issues, you can disable it using this method.
```

Comment:
Nit: "If this causes any issues" - describe that a NAT or network misconfiguration (i.e. shard-aware port 19042 not exposed to the external world) is a common cause.